dp-identity-data-migration/groupsdataextraction/group_extraction.go

```diff
@@ -0,0 +1,283 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"strings"
+
+	"time"
+
+	"github.com/ONSdigital/dp-zebedee-sdk-go/zebedee"
+)
+
+type config struct {
+	groups_filename,
+	groupusers_filename,
+	host,
+	pword,
+	user string
+}
+
+func readConfig() (conf config) {
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+
+		if pair[0] == "groups_filename" {
+			conf.groups_filename = pair[1]
+		}
+
+		if pair[0] == "groupusers_filename" {
+			conf.groupusers_filename = pair[1]
+		}
+
+		if pair[0] == "zebedee_user" {
+			conf.user = pair[1]
+		}
+		if pair[0] == "zebedee_pword" {
+			conf.pword = pair[1]
+		}
+		if pair[0] == "zebedee_host" {
+			conf.host = pair[1]
+		}
+	}
+
+	if conf.host == "" || conf.pword == "" || conf.user == "" || conf.groups_filename == "" || conf.groupusers_filename == "" {
+		fmt.Println("Please set Environment Variables ")
+		fmt.Println(conf.host)
+
+		os.Exit(1)
+	}
+
+	return conf
+}
+
+type group struct {
+	GroupName,
+	UserPoolId,
+	Description,
+	RoleArn,
+	Precedence,
+	LastModifiedDate,
+	CreationDate string
+}
+
+var group_header = group{
+	GroupName:        "group_name",
+	UserPoolId:       "user_pool_id",
+	Description:      "description",
+	RoleArn:          "role_arn",
+	Precedence:       "precedence",
+	LastModifiedDate: "last_modified_date",
+	CreationDate:     "creation_date",
+}
+
+type user_group_csv struct {
+	Username string
+	Groups   string
+}
+
+var header_user_groups = user_group_csv{
+	Username: "user_name",
+	Groups:   "groups",
+}
+
+func getgroups(zebCli zebedee.Client, s zebedee.Session) (grouplist zebedee.TeamsList, err error) {
```

Comment:
`zebCli.ListTeams(s)` can be called from line 170 (main) so this function can be removed.

Reply:
Rmoved function